data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/InvalidCloudEventInterceptorTest.java

```diff
@@ -294,7 +293,7 @@ private static ConsumerRecords<Object, CloudEvent> mockInvalidRecords() {
       ),
       new TopicPartition("t1", 1),
       List.of(new ConsumerRecord<>(
```

Comment:
This `ctor` is deprecated w/ 3.x, and we actually change the checksum here.

```
use one of the constructors without a `checksum` parameter. This constructor will be removed in Apache Kafka 4.0 (deprecated since 3.0).
```
see:
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java#L164-L173

Reply:
Yes, but inference doesn't work in this case, so I left the use of a deprecated constructor in this test

Reply:
ok, but the value is IGNORED, so `null` is OK... but misleading a bit too :sweat_smile:  semantically makes sense, but I did wonder why from actual to null (I get now that it's deprecated / ignored)